src/polytope/models/Problem.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass, field
+from typing import List
+
+from .ProblemStatement import ProblemStatement
+from .ProblemChecker import ProblemChecker
+from .ProblemValidator import ProblemValidator
+from .ProblemTest import ProblemPublicTest, ProblemTest
+from .ProblemSolution import ProblemSolution
+
+
+@dataclass
+class Problem:
+    """! A Polytope problem dataclass."""
+
+    """! Problem ID.
+
+    * It must be auto-generated by using UUID.
+    * It must be read-only.
+
+    TODO: Define UUID type and
+          set UUID generator as a default_factory.
+    """
+    _id: str
+
+    """! Problem name.
+
+    * It must be non-empty.
+    """
+    name: str
+
+    """! Draft note for the problem."""
+    note: str
+
+    """! Time limit in milliseconds.
+
+    * It must be positive.
+    """
+    time_limit_in_ms: int
+
+    """! Memory limit in mebibytes.
+
+    * It must be positive."""
+    memory_limit_in_mib: int
+
+    """! Problem tags.
+
+    * Tags must be distinct.
+    * A tag must be non-empty.
+    """
+    tags: List[str]
+
+    """! Problem owner."""
+    # owner: GithubUser
+
+    """! Problem statements."""
+    statements: List[ProblemStatement] = field(default_factory=list)
```

Comment:
(question) Why did you pick `List` as type of statements? Don't we have to extend it into map per language or format(html/tex/pdf...)?

Reply:
I have no ideas.
We do have to match the languages and formats with the statements, but it seems not a good idea to use the attributes of `ProblemStatement` as a key of `Dict`.

Would you suggest a good solution?

Reply:
@utilForever 
Thank you! I modified it so that there can be multiple owners.

https://github.com/Studio-Polytope/Polytope/blob/f043c6d87a2c3dadfd29cdf94c2fffeeb2fff8f7/src/polytope/models/Problem.py#L50-L56